src/main/java/org/pentaho/di/job/entries/spark/JobEntrySparkSubmit.java

```diff
@@ -542,6 +544,51 @@ public void check( List<CheckResultInterface> remarks, JobMeta jobMeta, Variable
     andValidator().validate( this, "className", remarks, putValidators( notBlankValidator() ) );
   }
 
+  /**
+   * Parse a string into arguments as if it were provided on the command line.
+   *
+   * @param commandLineString
+   *          A command line string.
+   * @param variableSpace
+   *          Context for resolving variable names. If {@code null}, no variable resolution will happen.
+   * @return List of parsed arguments
+   * @throws IOException
+   *           when the command line could not be parsed
+   */
+  public List<String> parseCommandLine( String commandLineString ) throws IOException {
```

Comment:
Do we not have a third-party library or existing PDI method to parse a command line?